src/tir/ir/data_layout.cc

```diff
@@ -323,7 +323,12 @@ inline Array<PrimExpr> TransformShape(const Array<PrimExpr>& src_shape,
       if (symbolic_var_set.count(i)) {
         result.push_back(tir::Any());
       } else {
-        result.push_back(ana.Simplify(tir::Substitute(rule, bind_map)));
+        auto sub = tir::Substitute(rule, bind_map);
+        if (sub.as<tir::AnyNode>()) {
+          result.push_back(tir::Any());
+        } else {
+          result.push_back(ana.Simplify(sub));
+        }
```

Comment:
Perhaps in this case we should go and fix Simplify for Any in the arith module and also add a testcase there

Reply:
Good point, let me get my head around the semantics and first order simplifications.

Reply:
Please followup :)

Reply:
I ended up avoiding the layout transformation on the dynamic dimension by specifying it from the input which unblocked me, making this a bit lower priority. I could still try to push this through if you prefer but thinking I will close for now and reopen at a later time.